core/templates/domain/objects/NumberWithUnitsObjectFactorySpec.ts

```diff
@@ -176,5 +191,54 @@ describe('NumberWithUnitsObjectFactory', () => {
         'SyntaxError: In "kg / m^(2)", "^" must be ' +
         'followed by a floating-point number');
     });
+
+    it('should create currency units', () => {
+      const createCurrencyUnitsSpy = spyOn(
+        nwuof.unitsFactory, 'createCurrencyUnits');
+      nwuof.createCurrencyUnits();
+      expect(createCurrencyUnitsSpy).toHaveBeenCalled();
+    });
+
+    it('should create NumberWithUnits object from dict', () => {
+      let numberWithUnitsObject = {
+        type: 'dummy-type',
+        real: 1,
+        fraction: {
+          isNegative: false,
+          wholeNumber: 2,
+          numerator: 1,
+          denominator: 3
+        },
+        units: [
+          {
+            unit: 'Kg',
+            exponent: 2
+          }
+        ]
+      };
+
+      let createdNumberWithUnits = nwuof.fromDict(numberWithUnitsObject);
+      expect(createdNumberWithUnits.toDict()).toEqual(numberWithUnitsObject);
+    });
+
+    it('should convert list to math.js compatible string', () => {
+      expect((new NumberWithUnits(
+        'real',
+        1,
+        new Fraction(
+          false, 2, 1, 3),
```

Comment:
What's going on here -- why is the 2 1/3 ignored?

(Please don't "fix" this by just removing the fraction part from this test, it seems like there should be an error raised in the main code if the constructor is given arguments that are not compatible? If so we should add that check and also test for it.)

Reply:
Should we also check the reverse, i.e. when we include a real number in fraction type then it should raise an error?

Reply:
I have added a check + tests for both. PTAL

Reply:
Looks better -- thanks!